x-pack/test/security_solution_cypress/cypress/tasks/assistant.ts

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DEFAULT_SYSTEM_PROMPT_NON_I18N } from '@kbn/security-solution-plugin/public/assistant/content/prompts/system/translations';
+import { TIMELINE_CHECKBOX } from '../screens/timelines';
+import { CLOSE_FLYOUT } from '../screens/alerts';
+import {
+  AI_ASSISTANT_BUTTON,
+  ASSISTANT_CHAT_BODY,
+  CHAT_ICON,
+  CHAT_ICON_SM,
+  CONNECTOR_SELECT,
+  CONNECTOR_SELECTOR,
+  CONVERSATION_TITLE,
+  EMPTY_CONVO,
+  WELCOME_SETUP,
+  NEW_CHAT,
+  CONVERSATION_SELECT,
+  FLYOUT_NAV_TOGGLE,
+  CONVERSATION_MESSAGE,
+  USER_PROMPT,
+  SUBMIT_CHAT,
+  CONVERSATION_MESSAGE_ERROR,
+  CLEAR_SYSTEM_PROMPT,
+  CHAT_CONTEXT_MENU,
+  CLEAR_CHAT,
+  CONFIRM_CLEAR_CHAT,
+  SYSTEM_PROMPT_SELECT,
+  SYSTEM_PROMPT,
+  CREATE_SYSTEM_PROMPT,
+  SYSTEM_PROMPT_TITLE_INPUT,
+  SYSTEM_PROMPT_BODY_INPUT,
+  CONVERSATION_MULTI_SELECTOR,
+  MODAL_SAVE_BUTTON,
+  ADD_QUICK_PROMPT,
+  QUICK_PROMPT_TITLE_INPUT,
+  QUICK_PROMPT_BODY_INPUT,
+  PROMPT_CONTEXT_SELECTOR,
+  QUICK_PROMPT_BADGE,
+  ADD_NEW_CONNECTOR,
+  SHOW_ANONYMIZED_BUTTON,
+  ASSISTANT_SETTINGS_BUTTON,
+} from '../screens/ai_assistant';
+import { TOASTER } from '../screens/alerts_detection_rules';
+
+export const openAssistant = (context?: 'rule' | 'alert') => {
+  if (!context) {
+    cy.get(AI_ASSISTANT_BUTTON).click();
+    return;
+  }
+  if (context === 'rule') {
+    cy.get(CHAT_ICON).should('exist');
```

Comment:
If you are adding this line to make the test more robust, is better to change `exist` to `be.visible`, since with the current solution, if the element is in the dom but not visible, that assertion will pass.